dode/client.go

```diff
@@ -0,0 +1,95 @@
+// Direct implementation of all required do.de DNS API endpoints
+
+package dode
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+const baseURl = "https://www.do.de/api"
+
+type ApiResponse struct {
```

Comment:
```suggestion
type APIResponse struct {
```

This is a very unimportant nit, but in Go, convention is to typically capitalize initialisms like API, DNS, etc. etc.